R/inside-the-house.R

```diff
@@ -0,0 +1,22 @@
+from_permitted_package <- function(env = parent.frame()) {
+  env <- topenv(env, globalenv())
+  if (!isNamespace(env)) {
+    return(FALSE)
+  }
+
+  nm <- getNamespaceName(env)
+  cat_line("attempt from: ", nm)
+  nm %in% c("gargle", "googledrive", "googlesheets4", "gmailr", "bigrquery")
+}
+
+check_permitted_package <- function(env = parent.frame()) {
+  if (!from_permitted_package(env)) {
+    msg <- paste(
+      "Resource is only available inside specific tidyverse packages.",
```

Comment:
Would it be better to frame this as "gargle project"? Since that's what you'll need to create in the cloud console?

Reply:
I don't quite follow. The user (or package author) who runs into this doesn't need to create a "gargle project" *per se*. They just need to bring their own API key or OAuth client id/secret pair. Is this just about proposing a project name for them?

Reply:
(And yes those things must live inside a GCP project). But the connection between the user's GCP project (and its assets around credentials) and gargle would only be a state of mind.

Reply:
I think I mean "google project" 😞